azkaban-web-server/src/main/java/azkaban/flowtrigger/database/JdbcFlowTriggerInstanceLoaderImpl.java

```diff
@@ -101,7 +101,8 @@
           + "project_json, flow_exec_id \n"
           + "FROM execution_dependencies JOIN (\n"
           + "SELECT trigger_instance_id FROM execution_dependencies WHERE trigger_instance_id not in (\n"
-          + "SELECT distinct(trigger_instance_id)  FROM execution_dependencies WHERE dep_status =  0 or dep_status = 4)\n"
+          + "SELECT distinct(trigger_instance_id)  FROM execution_dependencies WHERE dep_status ="
+          + "  0 or dep_status = 3)\n"
```

Comment:
Would it be better to pass the dependency status as the parameter into the SQL string? In case the enum value is changed in the future, we don't need to change the code here.

Reply:
hmm, thanks for the suggestion.  I considered it initially, but the status is very unlikely to be changed so i would rather to keep the sql as a constant string.

Reply:
It's ok to change code in the future. We do it all the time. 
The question is how easy it is to change. 
If it is an implementation detail, we can change them in the future relatively easily. In that case, I would prefer the simplest and most readable solution.

Reply:
What's the significance of the number 0 and 3? Can it be made more readable?

Reply:
updated